fhir-model/src/main/java/com/ibm/fhir/model/path/evaluator/FHIRPathEvaluator.java

```diff
@@ -242,22 +242,24 @@ private void reset() {
             return empty();
         }
 
-        private Collection<FHIRPathNode> is(List<ExpressionContext> arguments) {
+        private Collection<FHIRPathNode> is(Collection<ExpressionContext> arguments) {
             if (arguments.size() != 1) {
                 throw unexpectedNumberOfArguments(arguments.size(), "is");
             }
-            ExpressionContext typeName = arguments.get(0);
             Collection<FHIRPathNode> currentContext = getCurrentContext();
-            if (isSingleton(currentContext)) {
-                String identifier = typeName.getText();
-                FHIRPathType type = FHIRPathType.from(identifier);
-                if (type == null) {
-                    throw new IllegalArgumentException(String.format("Argument '%s' cannot be resolved to a valid type identifier", identifier));
-                }
-                FHIRPathNode node = getSingleton(currentContext);
-                if (type.isAssignableFrom(node.type())) {
-                    return SINGLETON_TRUE;
-                }
+            if (!isSingleton(currentContext)) {
+                throw new IllegalArgumentException(String.format("Input collection has %d items, but only 1 is allowed", currentContext.size()));
+            }
+            
+            ExpressionContext typeName = arguments.iterator().next();
+            String identifier = typeName.getText();
+            FHIRPathType type = FHIRPathType.from(identifier);
+            if (type == null) {
+                throw new IllegalArgumentException(String.format("Argument '%s' cannot be resolved to a valid type identifier", identifier));
+            }
+            FHIRPathNode node = getSingleton(currentContext);
+            if (type.isAssignableFrom(node.type())) {
+                return SINGLETON_TRUE;
             }
             return SINGLETON_FALSE;
```

Comment:
If I'm looking at this correctly 261 and 262 and 264 could be collapsed into a single expression?

Reply:
yeah, it could just be:
`return type.isAssignableFrom(node.type()) ? SINGLETON_TRUE : SINGLETON_FALSE;`

about the same to me, but i'm happy to make that change

Reply:
Just an observation, no change necessary.

Reply:
I do believe in the Knuth principle of a single return

Reply:
updated it as suggested